x-pack/plugins/security/common/licensing/license_service.ts

```diff
@@ -14,8 +14,8 @@ import type { SecurityLicenseFeatures } from './license_features';
 export interface SecurityLicense {
   isLicenseAvailable(): boolean;
   isEnabled(): boolean;
-  getType(): LicenseType | undefined;
   getFeatures(): SecurityLicenseFeatures;
+  hasAtLeast(licenseType: LicenseType): boolean | undefined;
```

Comment:
We _could_ coerce this to a boolean but this seemed like a better idea.

Reply:
Can you explain why you think this is a better idea? The [one place](https://github.com/legrego/kibana/blob/480395738e601a56d4f86c748b2de6f70a96260c/x-pack/plugins/features/server/feature_privilege_iterator/sub_feature_privilege_iterator.ts#L29) we are using this implicitly coerces `undefined` to `false`, so it seems that we do want coercion, it's just a matter of where it takes place.

IMO it makes more sense to explicitly coerce here, unless/until we have a need for this return type to be an optional boolean.

Reply:
`getType()`, which this supersedes, also returned undefined if the license service wasn't available. Not sure if any other consumers may need to differentiate in the future between "license unavailable" or "minimum license not met". Happy to change it though if you feel explicit coercion is better here, I was a bit on the fence about it anyway.

Reply:
I don't feel strongly enough to have you change it, I mostly wanted to understand your reasoning. I'm good either way 👍